### PR TITLE
Images by tag

### LINKED ISSUE #10: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [x] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [x] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [ ] Where in the code should we implement this functionality?
- [ ] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by jmaher409**
Closes #10 

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [X] I have replaced `X` above with the associated issue number.
- [X] I have changed the issue's label to "pull request".
- [x] I have assigned the designated reviewer to this PR.
- [x] I have set the appropriate label on this PR.
- [X] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -134,6 +144,35 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
 
     ImagesView.unstub(:new)
-    view_model_mock.unstub(:sort_images)
+    view_model_mock.unstub(:images)
+    view_model_mock.unstub(:tags)
+  end
+
+  test '.index links to index filtered on tag for each available tag' do
+    view_model_mock = mock
+    ImagesView.expects(:new).returns(view_model_mock)
```
Comment (bhargav265) [inline test/controllers/images_controller_test.rb]: Can we assert the params too which are being passed to the view model.
forexample, `ImagesView.expects(:new).with(x,y,z).returns(view_model_mock)`

File: app/controllers/images_controller.rb [hunk 2]
```diff
@@ -4,7 +4,8 @@ def new
   end
 
   def index
-    @view_model = ImagesView.new
+    tag = params[:tag]
```
[the remark this answers is not in the document]
Comment (jmaher409) [reply app/controllers/images_controller.rb]: done

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -134,6 +144,54 @@ class ImagesControllerTest < ActionDispatch::IntegrationTest
...
+    view_model_mock.unstub(:tags)
+  end
+
+  test '.index passes id to images_view on entry' do
+    view_model_mock = mock
+    ImagesView.expects(:new).with('foo').returns(view_model_mock)
+    view_model_mock.expects(:tags).returns([])
+    view_model_mock.expects(:images).returns([])
+
+    get "#{images_path}/?tag=foo"
```
Comment (bhargav265) [inline test/controllers/images_controller_test.rb]: This is okay. but a rails way of doing it is get images_path, params: {tag: 'x, y, z'}. By using this, you need not worry about the params getting formatted if they have a space or special charaters etc.

Comment (jmaher409) [reply test/controllers/images_controller_test.rb]: thats much cleaner. Fixing now

**Pull request merged**